gix-blame/tests/blame.rs

```diff
@@ -1,4 +1,58 @@
+use std::path::PathBuf;
+
+use gix_odb::pack::FindExt;
+use gix_ref::{file::ReferenceExt, store::WriteReflog};
+
 #[test]
 fn it_works() {
-    let _worktree = gix_testtools::scripted_fixture_read_only("make_blame_repo.sh").unwrap();
+    // TODO
+    // At a high level, what we want to do is the following:
+    //
+    // - get the commit that belongs to a commit id
+    // - walk through parents
+    //   - for each parent, do a diff and mark lines that don’t have a suspect (this is the term
+    //     used in `libgit2`) yet, but that have been changed in this commit
+    //
+    // The algorithm in `libgit2` works by going through parents and keeping a linked list of blame
+    // suspects. It can be visualized as follows:
+    //
+    // <---------------------------------------->
+    // <---------------><----------------------->
+    // <---><----------><----------------------->
+    // <---><----------><-------><-----><------->
+    // <---><---><-----><-------><-----><------->
+    // <---><---><-----><-------><-----><-><-><->
+
+    let worktree = fixture_path();
+
+    let store = gix_ref::file::Store::at(
+        worktree.join(".git"),
+        gix_ref::store::init::Options {
+            write_reflog: WriteReflog::Disable,
+            ..Default::default()
+        },
+    );
+    let odb = odb_at("");
+
+    let mut reference = gix_ref::file::Store::find(&store, "HEAD").unwrap();
```

Comment:
I think it's a great idea to use tests to get acquainted with the API a bit. There is also a `gix-traverse` crate which provides iterators (for the most part) that return commits in some traversal order. That is probably a viable input to the actual blame function which then has to do the work.

Reply:
Thanks for the hint, `gix-traverse` seems really useful!